### DISCUSSION TITLE
SideSetsAroundSubdomainGenerator & SideSetsBetweenSubdomainsGenerator

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#27331] [Q&A General] [answered]
All,
In order to study "SideSetsAroundSubdomainGenerator" and "SideSetsBetweenSubdomainsGenerator" to be used for an InterfaceKernels app such as "InterfaceDiffusion," I ran the model shown below and produced the same results. So, I would like to say both "SideSetsAroundSubdomainGenerator" and "SideSetsBetweenSubdomainsGenerator" can be used for InterfaceKernels apps.
Please confirm.  S. Thomas

[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 20
    ny = 20
    xmax = 1
    ymax = 1
  []
  [./subdomain1]
    input = gen
    type = SubdomainBoundingBoxGenerator
    bottom_left = '0.5 0 0'
    top_right = '1.0 1.0 0'
    block_id = 1
  [../]
  [./interface]
    type = SideSetsAroundSubdomainGenerator
    input = subdomain1
    block = 0
    normal = '1 0 0' # right
    new_boundary = 'interface'
  [../]
.
.
[InterfaceKernels]
  [./Interface]
    type = InterfaceDiffusion
    variable = 'u'
    neighbor_var = 'v'
    D = 1
    D_neighbor = 3
    boundary = 'interface'
  [../]
[]

or

  [./interface]
    input = subdomain1
    type = SideSetsBetweenSubdomainsGenerator
    primary_block = '0'
    paired_block = '1'
    new_boundary = 'master0_interface'
  [../]
.
.
[InterfaceKernels]
  [./Interface]
    type = InterfaceDiffusion
    variable = 'u'
    neighbor_var = 'v'
    D = 1
    D_neighbor = 3
    boundary = 'master0_interface'
  [../]
[]

**Comment by GiudGiud** [comment, marked as answer]
Hello

Yes they should both be able to produce valid sidesets that you can use downstream. 

The parameters are a little different they differ a bit in how they specify the sideset

Guillaume

**Reply by styyokuda** [reply]
Thank you, Guillaume, as always.
